src/workerd/api/http.c++

```diff
@@ -417,6 +418,143 @@ bool Headers::inspectImmutable() {
   return guard != Guard::NONE;
 }
 
+// -----------------------------------------------------------------------------
+// serialization of headers
+//
+// http-over-capnp.capnp has a nice list of common header names, taken from the HTTP/2 standard.
+// We'll use it as an optimization.
+//
+// Note that using numeric IDs for headers implies we lose the original capitalization. However,
+// the JS Headers API doesn't actually give the application any way to observe the capitalization
+// of header names -- it only becomes relevant when serializing over HTTP/1.1. And at that point,
+// we are actually free to change the capitalization anyway, and we commonly do (KJ itself will
+// normalize capitalization of all registered headers, and http-over-capnp also loses
+// capitalization). So, it's certainly not worth it to try to keep the original capitalization
+// across serialization.
+
+// If any more headers are added to the CommonHeaderName enum later, we should be careful about
```

Comment:
Might be a good idea to reconcile this against the most recent qpack spec used for http/3 https://www.rfc-editor.org/rfc/rfc9204.html#appendix-A

Reply:
Weird, this list seems different from the HTTP/2 list that I based http-over-capnp on. But if we're going to change anything I'd prefer to update http-over-capnp itself first.